### PR TITLE
upgrade everything

### PR DESCRIPTION
**Pull request by benjaminpjones**
idk if we want a more elegant upgrade strategy, but I figure we're due after one year.  I tested the usual scripts (`yarn build`, `yarn start`, `yarn launch`), but that's about it.

The biggest thing in here is TypeScript 5.  Looks like Mongo had a major version bump too, but I didn't see any breakages.

Motivation for the upgrade was mainly an attempt to resolve these deprecation errors:

Screenshot 2023-07-06 at 5 03 56 PM

### CODE REVIEW

File: packages/vue-client/src/components/boards/ParallelGoBoard.vue [hunk 1]
```diff
@@ -109,6 +109,6 @@ const board = computed(() => {
   <MulticolorGridBoard
     :board="board"
     :config="props.config"
-    @click="(pos) => positionClicked(pos)"
```
Comment (benjaminpjones) [inline packages/vue-client/src/components/boards/ParallelGoBoard.vue:112]: changed to address implicit any error

File: packages/vue-client/tsconfig.app.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@vue/tsconfig/tsconfig.web.json",
```
Comment (benjaminpjones) [inline packages/vue-client/tsconfig.app.json:2]: see [vuejs/tsconfig README: Migrating from TypeScript <5.0](https://github.com/vuejs/tsconfig#migrating-from-typescript--50)

File: packages/vue-client/tsconfig.config.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@vue/tsconfig/tsconfig.node.json",
```
Comment (benjaminpjones) [inline packages/vue-client/tsconfig.config.json:2]: see [vuejs/tsconfig README: Migrating from TypeScript <5.0](https://github.com/vuejs/tsconfig#migrating-from-typescript--50)

**Review by JonKo314 — APPROVED** [review]
Well, `yarn test` had no issue, so I think we're good to go. :+1:

**Pull request merged**